integration/test/geopm_test_launcher.py

```diff
@@ -128,7 +129,7 @@ def run(self, test_name, include_geopm_policy=True, add_geopm_args=[], enable_af
             argv = ['dummy', detect_launcher(), '--geopm-ctl', self._pmpi_ctl,
                                                 '--geopm-agent', self._agent_conf.get_agent(),
                                                 '--geopm-profile', test_name,
-                                                '--geopm-program-filter', self._app_conf.get_exec_path()]
+                                                '--geopm-program-filter', os.path.basename(self._app_conf.get_exec_path())]
```

Comment:
Should I apply something similar here as well? 
* Add a `app_conf.get_program_filter()` function, defaulting to exec path but allowing individual appconf implementations to override it https://github.com/geopm/geopm/pull/3062/files#diff-76dd7a1a9349bf0d8cf85212926694ed5006d12dcb6a8b64e04c04a4da7a42ea
* Use `get_program_filter()` in launch_util.py https://github.com/geopm/geopm/pull/3062/files#diff-343359e818df706352ab39f88c63ca674e14f518dcc4eacc1fbd88feea99520d

Or could we do something similar to those changes as part of this PR so it applies for all users of appconf?

Reply:
Chris and I discussed this and thought it would be a good idea to make the change in the launcher.py.  If I did it there, all these downstream usages would not have to change.  Let's discuss.